### PR TITLE
Added support to fetch slot app service detail

### PR DESCRIPTION
**Pull request by ShilpiRachna1**
AzureResourceFilterUtility.getAppDetails() is used to validate that the app service exists and to fetch resource group and type of the app service. Previously it did not take slotname as an input and so the validation for app service failed.

With this change,  slotname is added as an input to the method. If not null and not production, 
- the resource type is "Microsoft.Web/Sites/Slots" instead of "Microsoft.Web/Sites"
- the slotname is appended to the app name to get the resource name.

So, the request url to fetch resource would be changed to the following for a slot app service:
https://management.azure.com/subscriptions/{SubscriptionId}/resources?api-version={version}&$filter=resourceType eq 'Microsoft.Web/Sites/Slots' and name eq '{app name}/{slot name}'

NOTE: npm package [azure-actions-appservice-rest](https://www.npmjs.com/package/azure-actions-appservice-rest) v1.3.16 is published and consumed from https://github.com/Azure/webapps-deploy in private branch shilpirachna1/slotrel for testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "homepage": "https://github.com/microsoft/pipelines-appservice-lib#readme",
   "devDependencies": {
     "@types/jest": "^24.0.15",
-    "@types/node": "^12.6.8",
+    "@types/node": "^12.20.55",
```
Comment (kumaraksh1) [inline package.json:22]: why we bumped it up?

**Review by kumaraksh1 — APPROVED** [review]
LGTM !!

File: package.json (hunk 1 above)
Comment (ShilpiRachna1) [reply package.json:22]: To fix build errors. Can you please try it locally once

**Review by surenderssm — APPROVED** [review]
LGTM!

**Pull request merged**